src/clarity-angular/button-group/button-group.ts

```diff
@@ -0,0 +1,229 @@
+
+@Component({
+    selector: "clr-button-group",
+    templateUrl: "button-group.html",
+    providers: [ButtonInGroupService],
+    host: {
+        "[class.btn-group]" : "true"
+    }
+})
+export class ButtonGroup {
+    @ContentChildren(Button) buttons: QueryList<Button>;
+
+    constructor(
+        public buttonGroupNewService: ButtonInGroupService,
+        private elementRef: ElementRef) {}
+
+    inlineButtons: Button[] = [];
+    menuButtons: Button[] = [];
+
+    /**
+     * 1. Initializes the initial Button Group View
+     * 2. Subscribes to changes on the ContentChildren
+     *    in case the user content projection changes
+     */
+    ngAfterContentInit() {
+        this.initializeButtons();
+        this.buttonGroupNewService.changes.subscribe(button => this.rearrangeButton(button));
+        this.buttons.changes.subscribe(() => {
+            this.initializeButtons();
+        });
+    }
+
+    /**
+     * Moves the button into the other ViewContainer
+     * when an update is received.
+     *
+     * @param button
+     */
+    rearrangeButton(button: Button): void {
+        let fromView: Button[];
+        let toView: Button[];
+        if (button.inMenu) {
+            fromView = this.inlineButtons;
+            toView = this.menuButtons;
+        } else {
+            fromView = this.menuButtons;
+            toView = this.inlineButtons;
+        }
+        const index: number = fromView.indexOf(button);
+        if (index > -1) {
+            fromView.splice(index, 1);
+            const moveIndex = this.getMoveIndex(button);
+            if (moveIndex <= toView.length) {
+                toView.splice(moveIndex, 0, button);
+            }
+        }
+    }
+
+    /**
+     * Author: Eudes
+     *
+     * Finds the order of a button w.r.t other buttons
+     *
+     * @param buttonToMove
+     * @returns {number}
+     */
+    getMoveIndex(buttonToMove: Button): number {
+        let tempArr: Button[] = this.buttons.filter(button => (button.inMenu === buttonToMove.inMenu));
+        return tempArr.indexOf(buttonToMove);
+    }
+
+    /**
+     * Finds where each button belongs based on
+     * the ContentChildren
+     */
+    initializeButtons(): void {
+        let tempInlineButtons: Button[] = [];
+        let tempInMenuButtons: Button[] = [];
+        this.buttons.forEach((button) => {
+            if (button.inMenu) {
+                tempInMenuButtons.push(button);
+            } else {
+                tempInlineButtons.push(button);
+            }
+        });
+        this.inlineButtons = tempInlineButtons;
+        this.menuButtons = tempInMenuButtons;
+    }
+
+    /**
+     * Overflow Menu
+     *
+     */
+
+    //Indicates the position of the overflow menu
+    private _menuPosition: string;
+
+    get menuPosition(): string {
+        return this._menuPosition;
+    }
+
+    @Input("clrMenuPosition")
+    set menuPosition(pos: string) {
+        if (pos && (menuPositions.indexOf(pos) > -1)) {
```

Comment:
The setter was actually pretty simple before but now since we are using the popover logic, the switch case is needed. I will check with Jeeyun if we can simplify this.